### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by LeoFrachet**

### CODE REVIEW

File: gtfs.js [hunk 1]
```diff
@@ -11,6 +11,14 @@ const getters = require('./helpers/getters');
 const { importTable } = require('./helpers/import');
 const schema = require('./helpers/schema');
 
+/**
+ * Table-generic function to add items in a table of a GTFS.
+ *
+ * @param {Array<Object>}  items     Array of item to add to the GTFS.
```
Comment (annaelleduff) [inline gtfs.js]: items*

File: gtfs.js [hunk 2]
```diff
@@ -35,6 +43,16 @@ function addItems(items, tableName, gtfs) {
   }
 }
 
+/**
+ * Table-generic function to get an indexed table of a GTFS. The indexation depends of the table, and is defined in
+ * the schema (If possible, it is the unique id).
+ *
+ * @param  {string}           tableName Name of the table of the GTFS to get.
```
Comment (annaelleduff) [inline gtfs.js]: will it work if the table name is invalid?

File: gtfs.js [hunk 3]
```diff
@@ -35,6 +43,16 @@ function addItems(items, tableName, gtfs) {
   }
 }
 
+/**
+ * Table-generic function to get an indexed table of a GTFS. The indexation depends of the table, and is defined in
+ * the schema (If possible, it is the unique id).
```
Comment (annaelleduff) [inline gtfs.js]: what schema?

File: gtfs.js [hunk 4]
```diff
@@ -11,6 +11,14 @@ const getters = require('./helpers/getters');
 const { importTable } = require('./helpers/import');
 const schema = require('./helpers/schema');
 
+/**
+ * Table-generic function to add items in a table of a GTFS.
+ *
+ * @param {Array<Object>}  items     Array of item to add to the GTFS.
+ * @param {string}         tableName Name of the table of the GTFS in which the objects should be added.
```
Comment (annaelleduff) [inline gtfs.js]: will it work if the name of the table is not valid?

File: gtfs.js [hunk 5]
```diff
@@ -101,6 +143,14 @@ function setIndexedItems(indexedItems, tableName, gtfs) {
 }
 
 class Gtfs {
+  /**
+   * Constructor of the GTFS
+   *
+   * @param  {string}           path                           Path to the folder contains the GTFS text files.
```
Comment (annaelleduff) [inline gtfs.js]: path to the folder _that_ contains the GTFS...?

File: gtfs.js [hunk 6]
```diff
@@ -101,6 +143,14 @@ function setIndexedItems(indexedItems, tableName, gtfs) {
...
+   * @param  {Object|undefined} regexPatternObjectsByTableName Optional ad-hoc regex to fix the tables. See importTable.
```
Comment (annaelleduff) [inline gtfs.js]: to fix how? see what about importTable ?

File: gtfs.js [hunk 7]
```diff
@@ -120,12 +170,38 @@ class Gtfs {
   }
 
   /* io */
+  /**
+   * Async function exporting the GTFS at a specific path.
+   *
+   * @param  {string}   path     Path to the folder which will contains the GTFS. The folder will be created if needed.
```
Comment (annaelleduff) [inline gtfs.js]: contain*

File: helpers/import.js [hunk 8]
```diff
@@ -9,14 +9,47 @@ const eachWithLog = require('./logging_iterator_wrapper');
...
 const schema = require('./schema');
 
-exports.importTable = (gtfs, tableName, options) => {
-  options = options || {};
-  const indexKeys = options.indexKeys || schema.indexKeysByTableName[tableName];
+/**
+ *
+ * @param {Gtfs}   gtfs      The GTFS in which to import the table.
+ * @param {string} tableName The table of the name to import.
+ * @param {
```
[the remark this answers is not in the document]
Comment (LeoFrachet) [reply helpers/import.js]: webstorm does parse it, and give advices based on that. Indeed GitHub seems to not like it.

[the remark this answers is not in the document]
Comment (quentin-sommer) [reply helpers/import.js]: perfect

**Pull request merged**